src/vdbfusion/pybind/vdb_volume.py

```diff
@@ -113,11 +110,17 @@ def integrate(
             ], "origin/extrinsic must be a (3,) array or a (4,4) matrix"
 
             _points = vdbfusion_pybind._VectorEigen3d(points)
+            if colors is not None:
+                _colors = vdbfusion_pybind._VectorEigen3d(colors)
+            else:
+                _colors = vdbfusion_pybind._VectorEigen3d(np.zeros((0, 3), dtype=np.float64))
             if weighting_function is not None:
-                return self._volume._integrate(_points, extrinsic, weighting_function)
-            if weight is not None:
-                return self._volume._integrate(_points, extrinsic, weight)
-            self._volume._integrate(_points, extrinsic)
+                _weighting_function = weighting_function
+            elif weight is not None:
+                _weighting_function = lambda x: weight
+            else:
+                _weighting_function = lambda x: 1.0
```

Comment:
Have you tested this? In my experience, this makes the fusion pipeline extremely slow. When I looked into the problem, I found [this](https://pybind11.readthedocs.io/en/stable/advanced/cast/functional.html): 
![image](https://user-images.githubusercontent.com/21349875/220087270-43c0adb9-94f8-4e03-abfe-2032669a212b.png)

That's why I came up with this madness of lambdas in the pybind module.

Any better idea?

Reply:
Yeah I tested it. It slows it down slightly. I reverted it as close as possible to the old style.